src/set.cr

```diff
@@ -136,17 +136,17 @@ struct Set(T)
     @hash.has_key?(object)
   end
 
-  # Removes the *object* from the set and returns `self`.
+  # Deletes the *object* and returns it, otherwise returns nil.
   #
   # ```
   # s = Set{1, 5}
   # s.includes? 5 # => true
-  # s.delete 5
+  # s.delete 5    # => 5
   # s.includes? 5 # => false
   # ```
   def delete(object)
```

Comment:
```suggestion
  def delete(object) : T?
```

Reply:
Ha, I didn't expect you suggesting to add return types :smile: .
I am not sure it is a good idea, `Hash` doesn't have this type restrictions.

Reply:
I guess it is safe, more than the type restriction of `object` at least.